app/models/league.rb

```diff
@@ -1,5 +1,4 @@
 class League < ActiveRecord::Base
```

Comment:
first of all, I think this is definitely an improvement and I like the null object pattern. :+1:

but I'm worried if all this stuff here belongs into the league. 
Why does the league care about querying and updating the users? 
My first intend would be to move as much as possible out of here and write proper scopes for theses queries, if possible. 

furthermore, why isn't there at least a simple spec for `#update_badges` ? :p